### PR TITLE
remove allow-untyped-defs from torch/distributed/pipelining/_debug.py

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143871

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/distributed/pipelining/_debug.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
-# mypy: allow-untyped-defs
 # Copyright (c) Meta Platforms, Inc. and affiliates
+
 import torch
+from torch.fx.node import Argument
 
 
-def friendly_debug_info(v):
+def friendly_debug_info(v: object) -> Argument:
```
Comment (Skylion007) [inline torch/distributed/pipelining/_debug.py:7]: Umm, how is a `str` an Argument? Doesn't seem to match the typing info for it unless I am missing something

Comment (bobrenjc93) [reply torch/distributed/pipelining/_debug.py:7]: It's a `BaseArgumentTypes` in torch.fx.node

```
BaseArgumentTypes = Union[
    str,
    int,
    float,
    bool,
    complex,
    torch.dtype,
    torch.Tensor,
    torch.device,
    torch.memory_format,
    torch.layout,
    torch._ops.OpOverload,
    torch.SymInt,
    torch.SymBool,
    torch.SymFloat,
]
base_types = BaseArgumentTypes.__args__  # type: ignore[attr-defined]

Target = Union[Callable[..., Any], str]

Argument = Optional[
    Union[
        Tuple["Argument", ...],
        Sequence["Argument"],
        Mapping[str, "Argument"],
        slice,  # Slice[Argument, Argument, Argument], but slice is not a templated type in typing
        range,
        "Node",
        BaseArgumentTypes,
    ]
]
```

**Pull request closed without merging**